platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ComponentReportingTasksPlugin.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.diagnostics.internal.DiagnosticsTaskNames;
+
+/**
+ * Plugin that adds tasks to report on the deprecated software model components configured for the project.
+ *
+ * @since 9.0
+ */
+@Incubating
+abstract public class ComponentReportingTasksPlugin implements Plugin<Project> {
```

Comment:
Is there a reason you made these into two separate plugins? What use cases do users have for applying them separately?

Reply:
Well, `base-diagnostics` and `software-diagnostics` were split into two separate projects (and the reporting tasks split into to two separate plugins) in 8.13.  I presume that was for platform ownership purposes, although that's speculation on my part.  If so, I think it makes sense to keep them separate.

There's no reason why _users_ would apply these plugins - I can't imagine a realistic scenario where they would need these plugins, but would not also be applying a rule-based plugin.  We do apply them in certain tests where a `RuleSource` is not applied but we want the report in order to verify functionality, but thsoe are not "real" scenarios that a user might encounter.